src/main/java/io/github/thebusybiscuit/exoticgarden/PlantsListener.java

```diff
@@ -0,0 +1,415 @@
+            e.getBlock().getWorld().dropItemNaturally(e.getBlock().getLocation(), item);
+        }
+    }
+
+    @EventHandler
+    public void onInteract(PlayerInteractEvent e) {
+        if (e.getAction() != Action.RIGHT_CLICK_BLOCK) return;
+        if (e.getHand() != EquipmentSlot.HAND) return;
+        if (e.getPlayer().isSneaking()) return;
+
+        if (SlimefunPlugin.getProtectionManager().hasPermission(e.getPlayer(), e.getClickedBlock().getLocation(), ProtectableAction.BREAK_BLOCK)) {
+            ItemStack item = ExoticGarden.harvestPlant(e.getClickedBlock());
+
+            if (item != null) {
+                e.getClickedBlock().getWorld().playEffect(e.getClickedBlock().getLocation(), Effect.STEP_SOUND, Material.OAK_LEAVES);
+                e.getClickedBlock().getWorld().dropItemNaturally(e.getClickedBlock().getLocation(), item);
+            }
+        }
+    }
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onBlockExplode(BlockExplodeEvent e) {
+        e.blockList().removeAll(getAffectedBlocks(e.blockList()));
+    }
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onEntityExplode(EntityExplodeEvent e) {
+        e.blockList().removeAll(getAffectedBlocks(e.blockList()));
+    }
+
+    private Set<Block> getAffectedBlocks(List<Block> blockList) {
+        Set<Block> blocksToRemove = new HashSet<>();
+
+        for (Block block : blockList) {
+            ItemStack item = ExoticGarden.harvestPlant(block);
+
+            if (item != null) {
+                blocksToRemove.add(block);
+                block.getWorld().dropItemNaturally(block.getLocation(), item);
+            }
+        }
+
+        return blocksToRemove;
+    }
+
+    private void dropFruitFromTree(Block block) {
+        for (int x = -1; x < 2; x++) {
+            for (int y = -1; y < 2; y++) {
+                for (int z = -1; z < 2; z++) {
+                    // inspect a cube at the reference
+                    Block fruit = block.getRelative(x, y, z);
+                    if (fruit.isEmpty()) continue;
+
+                    Location loc = fruit.getLocation();
+                    SlimefunItem check = BlockStorage.check(loc);
+                    if (check == null) continue;
+
+                    for (Tree tree : ExoticGarden.getTrees()) {
+                        if (check.getID().equalsIgnoreCase(tree.getFruitID())) {
+                            BlockStorage.clearBlockInfo(loc);
+                            ItemStack fruits = check.getItem();
+                            fruit.getWorld().playEffect(loc, Effect.STEP_SOUND, Material.OAK_LEAVES);
+                            fruit.getWorld().dropItemNaturally(loc, fruits);
+                            fruit.setType(Material.AIR);
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+}
```

Comment:
You are adding this class randomly, it is not even used or was relocated before...